### PR TITLE
Issue 127

### PR DESCRIPTION
**Pull request by abramsymons**

### CODE REVIEW

File: web_services/foxx/brightid/schemas.js [hunk 1]
```diff
@@ -233,8 +233,8 @@ schemas = Object.assign({
 
   verificationGetResponse: joi.object({
     data: joi.object({
-      unique: joi.string().description("true if user is unique under given context"),
-      context: joi.string().description("the context name"),
+      unique: joi.string().description("true if user is unique under given application"),
```
Comment (abramsymons) [inline web_services/foxx/brightid/schemas.js:236]: `true if user is verified for an app`

File: web_services/foxx/brightid/schemas.js [hunk 2]
```diff
@@ -233,8 +233,8 @@ schemas = Object.assign({
...
+      application: joi.string().description("the application name"),
```
Comment (abramsymons) [inline web_services/foxx/brightid/schemas.js:237]: `app: joi.string().description("the key of app"),`

File: web_services/foxx/brightid/db.js [hunk 3]
```diff
@@ -653,9 +657,15 @@ function unusedSponsorships(app) {
...
 }
 
-function sponsor(user, app, timestamp) {
+function sponsor(user, appName, timestamp) {
+  // remove testblocks if exists
+  const app = getApp(appName);
+  const context = getContext(app.context);
+  const coll = db._collection(context.collection);
+  const contextIds = getContextIdsByUser(coll, user);
+  removeBlock(contextIds[0], 'sponsorship');
```
[the remark this answers is not in the document]
Comment (siftal) [reply web_services/foxx/brightid/db.js]: It's done

File: web_services/foxx/brightid/db.js [hunk 4]
```diff
@@ -695,6 +705,29 @@ function getState() {
   }
 }
 
+function addBlock(app, contextId, act) {
```
[the remark this answers is not in the document]
Comment (siftal) [reply web_services/foxx/brightid/db.js]: It's done

File: web_services/foxx/brightid/db.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (siftal) [reply web_services/foxx/brightid/db.js]: It's done

File: web_services/foxx/brightid/db.js [hunk 5]
```diff
@@ -695,6 +705,29 @@ function getState() {
...
+  testblocksColl.insert({
+    "app": app,
+    "contextId": contextId,
+    "act": act,
+    "timestamp": Date.now(),
+  });
```
[the remark this answers is not in the document]
Comment (siftal) [reply web_services/foxx/brightid/db.js]: it's done

File: web_services/foxx/brightid/db.js [hunk 6]
```diff
@@ -695,6 +705,29 @@ function getState() {
...
+}
+
+function removeBlock(contextId, act) {
```
[the remark this answers is not in the document]
Comment (siftal) [reply web_services/foxx/brightid/db.js]: it's done

File: web_services/foxx/brightid/index.js [hunk 7]
```diff
@@ -226,15 +225,28 @@ const handlers = {
...
+    if (! app) {
+      res.throw(404, 'app not found', {errorNum: APP_NOT_FOUND});
+    }
+
+    const context = db.getContext(app.context);
     if (! context) {
       res.throw(404, 'context not found', {errorNum: CONTEXT_NOT_FOUND});
     }
 
+    const blocks = db.getBlocks(appName, contextId);
```
[the remark this answers is not in the document]
Comment (siftal) [reply web_services/foxx/brightid/index.js]: it's done

File: web_services/foxx/brightid/index.js [hunk 8]
```diff
@@ -475,6 +506,24 @@ router.get('/state', handlers.stateGet)
   .summary("Get state of this node")
   .response(schemas.stateGetResponse);
 
+router.put('/testblocks/:app/:act/:contextId', handlers.testblocksPut)
+  .pathParam('app', joi.string().required().description("Unique name of the app"))
+  .pathParam('act', joi.string().required().description("The action name"))
+  .pathParam('contextId', joi.string().required().description('the contextId of user within the context'))
+  .queryParam('testingKey', joi.string().description('the testing private key of the app'))
+  .summary('Add a block on a verification to the user.')
+  .description('Add a block on a verification to the user.')
```
[the remark this answers is not in the document]
Comment (siftal) [reply web_services/foxx/brightid/index.js:516]: it's done

File: web_services/foxx/brightid/index.js [hunk 9]
```diff
@@ -475,6 +506,24 @@ router.get('/state', handlers.stateGet)
...
+  .response(null);
+
+router.delete('/testblocks/:app/:act/:contextId', handlers.testblocksDelete)
+  .pathParam('app', joi.string().required().description("Unique name of the app"))
+  .pathParam('act', joi.string().required().description("The action name"))
+  .pathParam('contextId', joi.string().required().description('the contextId of user within the context'))
+  .queryParam('testingKey', joi.string().description('the testing private key of the app'))
+  .summary('Add a block on a verification to the user.')
```
[the remark this answers is not in the document]
Comment (siftal) [reply web_services/foxx/brightid/index.js]: it's done

File: web_services/foxx/brightid/index.js [hunk 10]
```diff
@@ -475,6 +506,24 @@ router.get('/state', handlers.stateGet)
...
+  .description('Add a block on a verification to the user.')
```
[the remark this answers is not in the document]
Comment (siftal) [reply web_services/foxx/brightid/index.js]: it's done.

File: web_services/foxx/brightid/schemas.js (hunk 1 above)
Comment (siftal) [reply web_services/foxx/brightid/schemas.js]: it's done.

File: web_services/foxx/brightid/schemas.js (hunk 2 above)
Comment (siftal) [reply web_services/foxx/brightid/schemas.js]: it's done

File: web_services/foxx/brightid/index.js [hunk 11]
```diff
@@ -226,15 +225,28 @@ const handlers = {
   verificationGet: function(req, res){
     let unique = true;
     let contextId = req.param('contextId');
-    let contextName = req.param('context');
-    const verification = req.param('verification');
+    let appName = req.param('app');
```
[the remark this answers is not in the document]
Comment (siftal) [reply web_services/foxx/brightid/index.js]: it's done.

File: web_services/foxx/brightid/index.js [hunk 12]
```diff
@@ -339,10 +338,9 @@ const handlers = {
     res.send({
       data: {
         unique,
-        context: contextName,
+        application: appName,
```
[the remark this answers is not in the document]
Comment (siftal) [reply web_services/foxx/brightid/index.js]: it's done

File: web_services/foxx/brightid/index.js [hunk 13]
```diff
@@ -438,10 +470,9 @@ router.get('/operations/:hash', handlers.operationGet)
   .response(schemas.operationGetResponse)
   .error(404, 'Operation not found');
 
-router.get('/verifications/:context/:contextId', handlers.verificationGet)
-  .pathParam('context', joi.string().required().description('the context in which the user is verified'))
+router.get('/verifications/:app/:contextId', handlers.verificationGet)
+  .pathParam('app', joi.string().required().description('the application in which the user is verified'))
```
[the remark this answers is not in the document]
Comment (siftal) [reply web_services/foxx/brightid/index.js]: it's done

File: web_services/foxx/brightid/db.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (siftal) [reply web_services/foxx/brightid/db.js]: it's done

File: web_services/foxx/brightid/index.js [hunk 14]
```diff
@@ -475,6 +506,24 @@ router.get('/state', handlers.stateGet)
   .summary("Get state of this node")
   .response(schemas.stateGetResponse);
 
+router.put('/testblocks/:app/:act/:contextId', handlers.testblocksPut)
+  .pathParam('app', joi.string().required().description("Unique name of the app"))
```
[the remark this answers is not in the document]
Comment (siftal) [reply web_services/foxx/brightid/index.js]: it's done

File: web_services/foxx/brightid/index.js [hunk 15]
```diff
@@ -475,6 +506,24 @@ router.get('/state', handlers.stateGet)
...
+  .pathParam('act', joi.string().required().description("The action name"))
+  .pathParam('contextId', joi.string().required().description('the contextId of user within the context'))
+  .queryParam('testingKey', joi.string().description('the testing private key of the app'))
+  .summary('Add a block on a verification to the user.')
```
[the remark this answers is not in the document]
Comment (siftal) [reply web_services/foxx/brightid/index.js]: it's done

File: web_services/foxx/brightid/index.js [hunk 16]
```diff
@@ -475,6 +506,24 @@ router.get('/state', handlers.stateGet)
   .summary("Get state of this node")
   .response(schemas.stateGetResponse);
 
+router.put('/testblocks/:app/:act/:contextId', handlers.testblocksPut)
+  .pathParam('app', joi.string().required().description("Unique name of the app"))
+  .pathParam('act', joi.string().required().description("The action name"))
+  .pathParam('contextId', joi.string().required().description('the contextId of user within the context'))
+  .queryParam('testingKey', joi.string().description('the testing private key of the app'))
```
[the remark this answers is not in the document]
Comment (siftal) [reply web_services/foxx/brightid/index.js]: it's done

**Pull request merged**